validator/stateless_block_validator.go

```diff
@@ -307,11 +307,17 @@ func SetMachinePreimageResolver(ctx context.Context, mach *ArbitratorMachine, pr
 			if err != nil {
 				log.Error("Failed to deserialize DAS message", "err", err)
 			} else {
-				dasPreimage, err := das.Retrieve(ctx, seqMsg[40:])
+				keysetPreimage, err := das.KeysetFromHash(ctx, cert.KeysetHash[:])
 				if err != nil {
-					return fmt.Errorf("couldn't retrieve message from DAS %w", err)
+					log.Error("Couldn't get keyset from DAS", "err", err)
+				} else {
+					preimages[common.BytesToHash(cert.KeysetHash[:])] = keysetPreimage
+					dasPreimage, err := das.Retrieve(ctx, cert)
+					if err != nil {
+						return fmt.Errorf("couldn't retrieve message from DAS %w", err)
+					}
```

Comment:
Seems like this will error in a number of cases where `parseSequencerMessage` will not return an error. Any case that leads that function not calling `das.Retrieve`, but not returning an error should be able to verify successfully

Reply:
I'll do a careful side-by-side comparison of these and make sure they're aligned safely.